CRM/Financeextras/Form/Payment/Refund.php

```diff
@@ -0,0 +1,258 @@
+   */
+  public $_contributionId;
+
+  /**
+   * The contact id to find display name.
+   *
+   * @var int
+   */
+  public $contactId;
+
+  /**
+   * The contact name.
+   *
+   * @var string
+   */
+  public $contactName;
+
+  /**
+   * The refund reasons.
+   *
+   * @var array
+   */
+  public $reasons;
+
+  /**
+   * The total refund balance.
+   *
+   * @var int
+   */
+  public $totalRefundBalance;
+
+  /**
+   * The available amount.
+   *
+   * @var int
+   */
+  public $availableAmount;
+
+  /**
+   * The payment processor.
+   *
+   * @var array
+   */
+  public $paymentProcessor;
+
+  /**
+   * This function is called prior to building and submitting the form
+   */
+  public function preProcess() {
+    Civi::resources()->addStyleFile('io.compuco.financeextras', 'css/custom-civicrm.css');
+    Civi::resources()->addScriptFile('io.compuco.financeextras', 'js/custom.js');
+    $this->_contributionId = CRM_Utils_Request::retrieve('id', 'Positive', $this);
+    $entityFinancialTrxns = civicrm_api3('EntityFinancialTrxn', 'get', [
+      'sequential' => 1,
+      'entity_id' => $this->_contributionId,
+      'entity_table' => 'civicrm_contribution',
+      'options' => ['limit' => 0],
+    ])['values'];
+    $paymentInfos = [];
+    foreach ($entityFinancialTrxns as $entity) {
+      $trxn = civicrm_api3('FinancialTrxn', 'get', [
+        'id' => $entity['financial_trxn_id'],
+        'is_payment' => 1,
+      ])['values'];
+      foreach ($trxn as $i => $trxns) {
+        if (isset($trxns['payment_processor_id']) && $trxns['payment_processor_id'] != "") {
+          $chargeID = $trxns['trxn_id'];
+          $refundRows = \CRM_Stripe_BAO_StripeRefund::getRefundRow($chargeID);
+          $refunds = civicrm_api3('PaymentProcessor', 'refund_balance', [
+            'payment_processor_id' => $trxns['payment_processor_id'],
+            'charge_id' => $chargeID,
+          ])['values'];
+          $refundsBalances = $refunds[0]['data'];
+          $totalRefundBalance = 0;
+          foreach ($refundsBalances as $refundsBalance) {
+            if (!in_array($refundsBalance['id'], $refundRows)) {
+              $params = [
+                'refund_id' => $refundsBalance['id'],
+                'trxn_id' => $refundsBalance['charge'],
+                'payment_intent_id' => $refundsBalance['payment_intent'],
+                'amount' => $refundsBalance['amount'],
+                'currency' => $refundsBalance['currency'],
+                'status' => $refundsBalance['status'],
+                'reason' => $refundsBalance['reason'],
+                'created_date' => date('Y-m-d H:i:s', $refundsBalance['created']),
+                'created_by' => \CRM_Core_Session::getLoggedInContactID(),
+              ];
+
+              \CRM_Stripe_BAO_StripeRefund::create($params);
+
+            }
+            $totalRefundBalance = $refundsBalance['amount'] + $totalRefundBalance;
+          }
+          $totalRefundBalance = $totalRefundBalance / 100;
```

Comment:
I guess API should return Civi amount rather than Stripe amount, so we don't have to do calculation here.